### PR TITLE
Work towards enabling sub-domain support

### PR DESCRIPTION
**Pull request by mads-hartmann**
This isn't completed yet. I managed to get it to accept the config and create the buckets / route53 configs but it doesn't quite work yet: http://example.hatch.sh/

### CODE REVIEW

File: hatch/domain.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Domain(object):
```
Comment (skovhus) [inline hatch/domain.py:4]: I think now is a good time to add some py.test unit tests... Example: does this work for a .co.uk domain?

File: hatch/domain.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        self.main_domain = main_domain
+
+    @property
+    def full_domain(self):
+        if self.subdomain:
+            return "{}.{}".format(self.subdomain, self.main_domain)
+        return self.main_domain
+
+    @staticmethod
+    def parse(domainStr):
```
Comment (skovhus) [inline hatch/domain.py]: nitpicking `domain_str ` : )

File: hatch/domain.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+        if domainStr is None:
+            return None
+
+        parts = domainStr.split(".")
```
Comment (skovhus) [inline hatch/domain.py]: This doesn't really work for some TLDs. 

Maybe just use https://github.com/john-kurkowski/tldextract

Comment (mads-hartmann) [reply hatch/domain.py]: Haha yeah, I think this is a reasonable problem to pull down a dependency for

**Comment by mads-hartmann** [comment]
@skovhus http://example.hatch.sh/ seems to work now so it was probably just the DNS that was still propagating changes. Perhaps we should create a bit more logging during the initial creating of buckets, records etc. and print a notice saying that it might take a minute or two for the custom domain to be active :)

**Comment by skovhus** [comment]
Good idea with a log line when modifying things that takes time to propagate. : )

**Comment by skovhus** [comment]
```
Hatching...
It might take a few minutes before your changes are deployed... ☕️
```

**Comment by mads-hartmann** [comment]
It has been a few years. I think we can safely close this ;)

**Pull request closed without merging**